File: dotnet/01-commands/tests/SumOfDigitsKata/Number.cs
namespace SumOfDigitsKata
{
    public class Number
    {
        private long _number;
        
        public Number(long number)
        {
            _number = number;
        }
        
        public long DigitalRoot()
        {
            long result = 0;

            while (_number > 0)
            {
                result += _number % 10;
                _number = _number / 10;
            }

            if (result > 9)
            {
                Number number = new Number(result);
                result = number.DigitalRoot();
            }

            return result;
        }
    }
}

File: nodejs/01-commands/tests/sourcecode/SumOfDigitsKata/Number.js
class Number {
    constructor(number) {
        this._number = number;
    }

    digitalRoot() {
        let result = 0;
        let number = this._number;

        while (number > 0) {
            result += number % 10;
            number = Math.floor(number / 10);
        }

        if (result > 9) {
            const newNumber = new Number(result);
            result = newNumber.digitalRoot();
        }

        return result;
    }
}

module.exports = Number;